lib/pdk/util/template_uri.rb

```diff
@@ -207,14 +207,21 @@ def self.templates(opts)
       end
 
       # @returns String
-      def self.default_template_ref
+      def self.default_template_ref(uri = nil)
+        uri ||= default_template_uri
+        uri = new(uri) unless uri.is_a?(self)
+
         if PDK::Util.development_mode?
           'master'
         else
-          PDK::TEMPLATE_REF
+          uri.default? ? PDK::TEMPLATE_REF : 'master'
         end
       end
 
+      def default_template_ref
```

Comment:
Hrmmm well..this is a little confusing.  class vs instance name with different method signatures :-(

Reply:
Yeah, we don't really need the instance method, so i've removed it